test/formatter_test.cc

```diff
@@ -0,0 +1,163 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include "iceberg/util/formatter.h"
+
+#include <format>
+#include <map>
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include <gtest/gtest.h>
+
+#include "iceberg/statistics_file.h"
+
+namespace iceberg {
+
+// Tests for the std::format specializations
+TEST(FormatterTest, VectorFormat) {
+  std::vector<int> empty;
+  EXPECT_EQ("[]", std::format("{}", empty));
+
+  std::vector<int> nums = {1, 2, 3, 4, 5};
+  EXPECT_EQ("[1, 2, 3, 4, 5]", std::format("{}", nums));
+
+  std::vector<std::string> names = {"Alice", "Bob", "Charlie"};
+  EXPECT_EQ("[Alice, Bob, Charlie]", std::format("{}", names));
+}
+
+TEST(FormatterTest, MapFormat) {
+  std::map<std::string, int> empty;
+  EXPECT_EQ("{}", std::format("{}", empty));
+
+  std::map<std::string, int> ages = {{"Alice", 30}, {"Bob", 25}, {"Charlie", 35}};
+  EXPECT_EQ("{Alice: 30, Bob: 25, Charlie: 35}", std::format("{}", ages));
+}
+
+TEST(FormatterTest, UnorderedMapFormat) {
+  std::unordered_map<std::string, double> empty;
+  EXPECT_EQ("{}", std::format("{}", empty));
+
+  std::unordered_map<std::string, double> scores = {
+      {"Alice", 95.5}, {"Bob", 87.0}, {"Charlie", 92.3}};
+  std::string str = std::format("{}", scores);
+  EXPECT_TRUE(str.find("Alice: 95.5") != std::string::npos);
+  EXPECT_TRUE(str.find("Bob: 87") != std::string::npos);
+  EXPECT_TRUE(str.find("Charlie: 92.3") != std::string::npos);
+}
+
+TEST(FormatterTest, NestedContainersFormat) {
+  std::vector<std::map<std::string, int>> nested = {{{"a", 1}, {"b", 2}},
+                                                    {{"c", 3}, {"d", 4}}};
+
+  EXPECT_EQ("[{a: 1, b: 2}, {c: 3, d: 4}]", std::format("{}", nested));
+
+  std::map<std::string, std::vector<int>> nested_map = {
+      {"primes", {2, 3, 5, 7, 11}}, {"fibonacci", {1, 1, 2, 3, 5, 8, 13}}};
+  std::string result = std::format("{}", nested_map);
+  EXPECT_TRUE(result.find("primes") != std::string::npos);
```

Comment:
Note that you can do `EXPECT_THAT(result, testing::HasSubstr("primes"));` which will also give a much better error on failure